demos/smart_classroom_demo/src/action_detector.cpp

```diff
@@ -131,26 +131,34 @@ DetectedActions ActionDetection::fetchResults() {
     const auto loc_blob_name = new_network_ ? config_.new_loc_blob_name : config_.old_loc_blob_name;
     const auto det_conf_blob_name = new_network_ ? config_.new_det_conf_blob_name : config_.old_det_conf_blob_name;
 
-    const cv::Mat priorbox_out =
-        new_network_
-          ? cv::Mat()
-          : cv::Mat(ieSizeToVector(request->GetBlob(config_.old_priorbox_blob_name)->getTensorDesc().getDims()),
-                    CV_32F, request->GetBlob(config_.old_priorbox_blob_name)->buffer());
-
+    LockedMemory<void> locBlobMapped = as<MemoryBlob>(request->GetBlob(loc_blob_name))->rwmap();
     const cv::Mat loc_out(ieSizeToVector(request->GetBlob(loc_blob_name)->getTensorDesc().getDims()),
-                          CV_32F, request->GetBlob(loc_blob_name)->buffer());
+                          CV_32F, locBlobMapped.as<float*>());
 
+    LockedMemory<const void> detConfBlobMapped = as<MemoryBlob>(request->GetBlob(det_conf_blob_name))->rmap();
     const cv::Mat main_conf_out(ieSizeToVector(request->GetBlob(det_conf_blob_name)->getTensorDesc().getDims()),
-                                CV_32F, request->GetBlob(det_conf_blob_name)->buffer());
+                                CV_32F, detConfBlobMapped.as<float*>());
 
     std::vector<cv::Mat> add_conf_out;
     for (int glob_anchor_id = 0; glob_anchor_id < num_glob_anchors_; ++glob_anchor_id) {
         const auto& blob_name = glob_anchor_names_[glob_anchor_id];
+        LockedMemory<const void> blobMapped = as<MemoryBlob>(request->GetBlob(blob_name))->rmap();
         add_conf_out.emplace_back(ieSizeToVector(request->GetBlob(blob_name)->getTensorDesc().getDims()),
-                                  CV_32F, request->GetBlob(blob_name)->buffer());
+                                  CV_32F, blobMapped.as<float*>());
```

Comment:
This also creates a dangling pointer once `blobMapped` goes out of scope.

Reply:
I've fixed the problem, but I'm still a bit unsure whether I should pass LockedMemory objects to GetDetections function or even make them global variables. 

loc_out, main_conf_out, priorbox_out, add_conf_out are used in GetDetections, but this is outside the scope in which LockedMemory objects were created.

Reply:
> loc_out, main_conf_out, priorbox_out, add_conf_out are used in GetDetections, but this is outside the scope in which LockedMemory objects were created.

Technically true, but irrelevant. The problem is not scope per se, but object lifetime. To take `locBlobMapped` as an example, it's destroyed at the end of the function, and since `GetDetections`  completes _before_ the end of the function, it's okay for it to access `loc_out`. 

When I said "once `blobMapped` goes out of scope", it was a shorthand for "once  `blobMapped` is destroyed at the end of the block".